### PR TITLE
Add RabbitMq Support

### LINKED ISSUE #85: Feeding NBXplorer notifications to RabbitMQ
Hi! I am building a worker (.net core console) that uses NBXplorer.Client to listen for NBXplorer notifications, process each event and send them over to RabbitMQ.  

Should I be worried about buffering those events while I am processing them, or the client will take care of that for me? The assumption is that I'm subscribing to thousands of newTransaction and NewBlock events. 

Thanks!

### PR DESCRIPTION
**Pull request by farukterzioglu**
New transaction and new block notifications can be pushed to RabbitMq exchanges. 
Closes #85

### CODE REVIEW

**Comment by NicolasDorier** [comment]
tests do not pass

**Comment by farukterzioglu** [comment]
> tests do not pass
tests require RabbitMq to be running with docker-compose.
No idea how to pass tests in AppVeyor.

**Comment by NicolasDorier** [comment]
AppVeyor have docker I think, so you can probably just run it before running the tests.

**Comment by farukterzioglu** [comment]
Will you configure AppVeyor @NicolasDorier 
Or are there anything I can do? 

I can deactivate RabbitMq tests by unsetting host name under test configs. that way unit tests will pass but RabbitMq related tests won't run.

**Comment by NicolasDorier** [comment]
either disable the rabbitmq tests for appveyor, or add docker to appveyor. This is the .appveyor.yml file, there is nothing I need to do that you can't.

Note that if you use RabbitMQ, you probably want appveyor test it, because I regularily break code and the only thing that keep me notified is appveyor.

File: docker-compose.regtest.yml [hunk 1]
```diff
@@ -14,14 +14,24 @@ services:
...
       NBXPLORER_BTCRPCURL: http://bitcoind:43782/
       NBXPLORER_BTCNODEENDPOINT: bitcoind:39388
+<<<<<<< HEAD
+      NBXPLORER_rmqhost: rabbitmq
+      NBXPLORER_rmqvirtual: / 
+      NBXPLORER_rmquser: guest
+      NBXPLORER_rmqpass: guest
+      NBXPLORER_rmqtranex: NewTransaction
+      NBXPLORER_rmqblockex: NewBlock
+=======
```
Comment (NicolasDorier) [inline docker-compose.regtest.yml]: fix the conflict

Comment (farukterzioglu) [reply docker-compose.regtest.yml]: fixed

**Comment by farukterzioglu** [comment]
Tests were passing until updating to .Net Core 3.1
Can be seen in here https://ci.appveyor.com/project/farukterzioglu/nbxplorer/builds/29329565
with commit https://github.com/farukterzioglu/NBXplorer/commit/9edeeea989204dcbd74933bf2d2613045518a36c

But AppVeyor can't build .Net Core 3.1

**Comment by NicolasDorier** [comment]
ah fuck @farukterzioglu let's wait a bit that appveyor update then :(

**Comment by NicolasDorier** [comment]
oh actually I will merge, if the tests passed.

**Comment by NicolasDorier** [comment]
I commented the tests for rabitmq, make the CI fail.
Please fix it or there is big chance I will eventually break it.

**Comment by farukterzioglu** [comment]
I filtered RabbitMq tests on CircleCI. They only run on AppVeyor. 
Tests should pass now. 

If you break something, only AppVeyor will tell. Is it ok?

**Pull request merged**